internal/operands/metrics/resources.go

```diff
@@ -16,6 +16,133 @@ const (
 	componentAlertLabelValue = "ssp-operator"
 )
 
+type MetricDesc struct {
```

Comment:
So if I understand correctly we define MetricDesc then we cast it to metric in one place, and build a list of those.
In another place we go over MetricDesc and create promv1.Rules out of them.
I feel this is overly complicated for what we are doing here. We basically never use MetricDesc for anything directly.
Perhaps your idea is to make it easier to add metric rules as in the Open Close principle? If so, I like the idea, however it will be a bit confusing because the summary is actually part of the annotations. So either lets make the MetricDesc structure have "annotations" and in them the "summary", or maybe we can just promv1.Rule?
I would go with just having promv1.Rule